### ISSUE TITLE
Question: should I call RetryConfig.of(...) very frequently?

### ISSUE THREAD

**Issue by sming** [resilience4j/resilience4j#1086]
Resilience4j version:v1.5.0

Java version:11

TL;DR: Question: `resilience4j` code examples use `RetryConfig.of(...)` but in production code, that's making millions of requests and hence calling this function very frequently, is it a good idea? 

I ask because : 

1. each call to `RetryConfig.of(...)` creates a `new RetryImpl(...)` object which will add GC pressure
1. [this comment for the Retry interface](https://github.com/resilience4j/resilience4j/blob/master/resilience4j-retry/src/main/java/io/github/resilience4j/retry/Retry.java#L37) states: `A Retry instance is thread-safe can be used to decorate multiple requests.`
    - this makes me think that I should *not* be calling `RetryConfig.of(...)` for each request

This leads me to question B, which is almost certainly affected by the answer to the above: 

- what is a good `name` to pass to `RetryConfig.of(...)`? (Presumably a static string if we call `RetryConfig.of(...)` once, or something dynamic for each request if not?)

Many thanks,
Peter / sming

**Comment by RobWin** [comment]
Hi,
you can create a Retry instance once and reuse it everywhere. You don't need new Retry instance per call/request.

**Comment by sming** [comment]
Great. Thanks for the lightning fast response Robert!

Cheers, Pete

**Comment by sming** [comment]
question answered promptly 👍🏻
